modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/DateTimeObjectFieldBusinessType.java

```diff
@@ -173,21 +174,26 @@ public Set<String> getUnmodifiableObjectFieldSettingsNames() {
 	}
 
 	@Override
-	public Object getValue(
+	public Timestamp getValue(
 			ObjectField objectField, long userId, Map<String, Object> values)
 		throws PortalException {
 
-		String value = String.valueOf(
-			ObjectFieldBusinessType.super.getValue(
-				objectField, userId, values));
+		Object value = ObjectFieldBusinessType.super.getValue(
+			objectField, userId, values);
 
 		if (Validator.isNull(value)) {
 			return null;
 		}
 
+		if (value instanceof Date) {
+			Date date = (Date)value;
+
+			return new Timestamp(date.getTime());
+		}
+
 		return _getTimestamp(
```

Comment:
@dannielraposo I like this :+1: Just checking whether there are valid use cases where the value would be a string?